pkg/cloud/azure/services/networkinterfaces/networkinterfaces.go

```diff
@@ -158,7 +158,7 @@ func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
 	if len(nicSpec.ApplicationSecurityGroupNames) > 0 {
 		asgSvc := applicationsecuritygroups.NewService(s.Scope)
 
-		groups := make([]network.ApplicationSecurityGroup, len(nicSpec.ApplicationSecurityGroupNames))
```

Comment:
can we instead `var groups []string` rather than initialise to empty?

we could also `make([]network.ApplicationSecurityGroup, 0, len(nicSpec.ApplicationSecurityGroupNames))` but I'd still prefer to initialise to nil as above
It'd be actually nice to put the whole block into a function and add a unit actually, not a blocker for this PR though

Reply:
@enxebre 
> can we instead var groups []string rather than initialise to empty?
I was trying to follow: 
https://github.com/openshift/cluster-api-provider-azure/blob/d1d00cae0b7d5480c93a3793fbd6711daefbe862/pkg/cloud/azure/services/networkinterfaces/networkinterfaces.go#L104

but `var groups []network.ApplicationSecurityGroup` also seemed fine, so change to that.

> It'd be actually nice to put the whole block into a function and add a unit actually, not a blocker for this PR though

i can try to do that as follow-up